docs/source-2.0/additional-specs/smoke-tests.rst

```diff
@@ -0,0 +1,194 @@
+        that can be represented in plain text as the plain text representation
+        (for example, use "foo" and not "Zm9vCg==").
+    * - vendorParams
+      - ``document``
+      - Defines vendor-specific parameters that are used to influence the
+        request. For example, some vendors might utilize environment variables,
+        configuration files on disk, or other means to influence the
+        serialization formats used by clients or servers.
+
+        If a ``vendorParamsShape`` is set, these parameters MUST be compatible
+        with that shape's definition.
+    * - vendorParamsShape
+      - ``string``
+      - A shape to be used to validate the ``vendorParams`` member contents.
+
+        If set, the parameters in ``vendorParams`` MUST be compatible with this
+        shape's definition.
+    * - expect
+      - :ref:`Expectation <Expectation-union>`
+      - **Required**. Defines the response that is expected from the service
+        call. This can be either a successful response, an error message, or
+        a specific error response.
+    * - tags
+      - ``[string]``
+      - Attaches a list of tags that can be used to categorize and group
+        test cases. If a test case uses a feature that requires special
+        configuration, it should be tagged.
+
+.. _Expectation-union:
+
+Expectation
+-----------
+
+A union describing the different kinds of expectations that can be made for a
+test case. As it is a union, exactly one member must be set.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 25 65
+
+    * - Property
+      - Type
+      - Description
+    * - success
+      - ``Unit``
+      - Indicates that the call is expected to not throw an error. No other
+        assertions are made about the response.
+    * - failure
+      - :ref:`FailureExpectation <FailureExpectation-struct>`
+      - Indicates that the call is expected to throw an error.
+
+.. _FailureExpectation-struct:
+
+FailureExpectation
+~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 25 65
+
+    * - Property
+      - Type
+      - Description
+    * - errorId
+      - ``string``
+      - Indicates that the call is expected to throw a specific type of error
+        matching the targeted shape. If not specified, the error can be of
+        any type.
+
+Smoke tests example
+===================
+
+The following example defines smoke test cases for an operation which should
+return a specific ``InvalidMessageError`` response when given an invalid input,
+and return a successful response otherwise.
+
+.. code:: smithy
+
+    $version: "2"
+
+    namespace smithy.example
+
+    use smithy.test#smokeTests
+
+    @smokeTests(
+        [
+            {
+                id: "FooSuccess"
+                params: {bar: "2"}
+                expect: {
+                    success: {}
+                }
+            }
+            {
+                id: "FooInvalidMessageError"
+                params: {bar: "föö"}
```

Comment:
nit: here would be a good place to have a `description` member, maybe saying why the message is considered invalid